### ISSUE TITLE
Add WikiNav to report

### ISSUE THREAD

**Issue by Sadads** [wikimedia/popularpages#20]
So the research team at the Wikimedia Foundation recently produced: https://wikinav.toolforge.org/ -- it would be really great to add this as an additional link/column for the sources of traffic for each of the articles. All that would be required is to put a link with the pagetitle name in the url i.e. https://wikinav.toolforge.org/?language=en&title=Climate_change_in_Nepal

**Comment by Sadads** [comment]
@MusikAnimal  -- I suspect this isn't too complicated, but let me know.

**Comment by Sadads** [comment]
Yeah, it only runs on historical data right now -- Martin Gerlach and folks are working to make it more of a service I think, but the visualizations are only useful once you hit a certain threshold of volume.

**Comment by MusikAnimal** [comment]
Do you know how often it updates? I see it is still showing data for August 2021, while pageview data for September is now available. My only concern here is expectations from users. Either coming from Topviews or from Popular Pages, they might expect data within the same time frame.

It also doesn't seem to work all the time, for instance https://wikinav.toolforge.org/?language=en&title=Hanksy. I'm guessing any page with low pageviews probably is missing clickstream data (for privacy reasons), hence why you're only interested in advertising this tool for popular pages?

Another thing to note: Popular Pages reports pageviews for the target page + all of its redirects. So users will most always get different numbers for the "actual number of incoming pageviews" in WikiNav. Not a big deal, but the discrepancy could be misleading.

**Comment by martingerlach** [comment]
@MusikAnimal trying to answer some of your questions on WikiNav:

- it only shows data from one month (the latest available)
- it automatically updates one a month to fetch the new data from the clickstream-dump
- you can get the list of supported languages and the latest snapshot via this call: https://wikinav.wmcloud.org/api/v1/latest/meta
- there is an open task to add more languages to the clickstream (and thus to WikiNav) https://phabricator.wikimedia.org/T289532
- the clickstream data contains only those links which were clicked more than 10 times in one month (due to privacy); thus pages with few pageviews will not yield any result in the WikiNav tool.

Let me know if you have further questions.
Thanks in advance for giving visibility to this tool (potentially); let me know how I can be of help.

**Comment by MusikAnimal** [comment]
@martingerlach That all makes sense to me. Thanks for the detailed reply!

@Sadads I am certainly fine with the concept of adding a link to WikiNav in the report, but there are a few issues. First, per privacy reasons that Martin explained, WikiNav will apparently not work for many pages that are listed in Popular Pages reports. I worry people will be confused by that, and bug reports will come in on CommTech's plate.

The bigger issue right now is that the bot also runs on Arabic Wikipedia, and that is not supported by WikiNav (yet). So we'd need to conditionally show it for English Wikipedia only. This kind of logic is not something I think belongs in the bot's code, which is meant to be wiki-agnostic. Instead, I'm thinking we could introduce something akin to [interface messages](https://www.mediawiki.org/wiki/Help:System_message), whereby each community (enwiki in this case) can customize the links for themselves. Perhaps even we could have customization on a per-WikiProject basis, so we can remove the links for smaller WPs whose articles don't receive a lot of traffic and hence wouldn't benefit from WikiNav.

In lieu of a Phabricator project for Popular Pages bot, I've created a new GitHub issue at #21 to implement the above idea. I will have to be honest though, I am currently swamped with other work as is Community Tech, so I feel this may be low-priority at this point. At any rate, I think the ability to customize the links is a great idea and worthwhile, when time allows to implement it. I've added a "help wanted" tag to the issue in the off-chance a volunteer (or either of you) wants to take a stab at it. I hope this is satisfactory for now, and apologies I can't do more at the moment!
